keps/sig-node/0000-apparmor-ga/README.md

```diff
@@ -0,0 +1,685 @@
+    ...
+    // The AppArmor options to use by this container. If AppArmor options are
+    // provided at both the pod & container level, the container options
+    // override the pod options.
+    // +optional
+    AppArmor  *AppArmorProfile
+    ...
+}
+
+// Only one profile source may be set.
+// +union
+type AppArmorProfile struct {
+    // +unionDescriminator
+    Type AppArmorProfileType
+
+    // Load a profile defined on the node.
+    // The profile must be available on the node to work.
+    // The length of the profile is limited to 253 characters.
+    // The `Type` in this struct has to be set to `AppArmorProfileLocalhost`.
+    // +optional
+    LocalhostProfile *string
+}
+
+type AppArmorProfileType string
+
+const (
+    AppArmorProfileTypeUnconfined     AppArmorProfileType = "Unconfined"
+    AppArmorProfileTypeRuntimeDefault AppArmorProfileType = "RuntimeDefault"
+    AppArmorProfileTypeLocalhost      AppArmorProfileType = "Localhost"
+)
+```
+
+This API makes the options more explicit and leaves room for new profile sources
+to be added in the future (e.g. Kubernetes predefined profiles or ConfigMap
+profiles) and for future extensions, such as defining the behavior when a
+profile cannot be set.
+
+##### Localhost Profile
+
+This KEP proposes we GA LocalhostProfile as the only source of user-defined
+profiles at this point. User-defined profiles are essential for users to realize
+the full benefits out of AppArmor, allowing them to decrease their attack
+surface based on their own workloads.
+
+###### Updating AppArmor profiles
+
+AppArmor profiles are applied at container creation time. Therefore, updating
+the file contents of a profile on disk after that point will not cause any
+changes to the running containers that are using it, but will apply the updated
+profile on container restart. Note that amending such files is not recommended
+and may cause containers to fail on next restart, in the case of the new profile
+being more restrictive, invalid or the file no longer being present on disk.
+
+Currently, users have no way to tell whether their physical profiles have been
+deleted or modified. This KEP proposes no changes to the existing functionality.
+
+The recommended approach for rolling out changes to AppArmor profiles is to
+always create _new profiles_ instead of updating existing ones. Create and
+deploy a new version of the existing Pod Template, changing the profile name to
+the newly created profile. Redeploy, once working delete the former Pod
+Template. This will avoid disruption on in-flight workloads.
+
+The current behavior lacks features to facilitate the maintenance of AppArmor
+profiles across the cluster. Two examples being: 1) the lack of profile
+synchronization across nodes and 2) how difficult it can be to identify that
+profiles have been changed on disk, after pods started using it. However, given
+its current "pseudo-GA" state, we don't want to change it with this KEP.
```

Comment:
The plan is to integrate it into the seccomp-operator, right? I think it's worth mentioning here (with the caveat that it's out of scope for this kep)

Reply:
Yes, I mentioned it in another sentence :+1: